[AArch64] Lower aarch64_neon_saddlv via SADDLV nodes.

davemgreen opened the pull request:
This mirrors what GISel already does, extending the existing lowering of aarch64_neon_saddlv/aarch64_neon_uaddlv to SADDLV/UADDLV. This allows us to remove some tablegen patterns, and provides a little nicer codegen in places as the nodes represent the result being in a vector register correctly.

SamTebbs33 commented on llvm/lib/Target/AArch64/AArch64ISelLowering.cpp:
```diff
@@ -6097,20 +6097,24 @@ SDValue AArch64TargetLowering::LowerINTRINSIC_WO_CHAIN(SDValue Op,
+                                OpVT == MVT::v8i16 || OpVT == MVT::v4i16)) ||
+         (ResVT == MVT::i64 && (OpVT == MVT::v4i32 || OpVT == MVT::v2i32))) &&
+        "Unexpected aarch64_neon_u/saddlv type");
+    // In order to avoid insert_subvector, used v4i32 than v2i32.
```
Should this be "use v4i32 rather than v2i32"?

SamTebbs33 reviewed (approved):
Very nice.

davemgreen committed:
[AArch64] Lower aarch64_neon_saddlv via SADDLV nodes.

This mirrors what GISel already does, extending the existing lowering of
aarch64_neon_saddlv/aarch64_neon_uaddlv to SADDLV/UADDLV. This allows us to
remove some tablegen patterns, and provides a little nicer codegen in places as
the nodes represent the result being in a vector register correctly.

davemgreen commented:
Cheers for the review.
